### ISSUE TITLE
datepicker-popup fails to apply custom classes correctly.  Fails to run digest.

### ISSUE THREAD

**Issue by dk8996** [angular-ui/bootstrap#5255]
Take a look at the two examples, one is a pop-up and the other one is inline.  The inline calendar works fine -- digest and custom-classes are applied while pop-up doesn't work.  The digest should be triggered by this line `$scope.dt = null;` but isn't in the datepicker-popup.

http://plnkr.co/edit/4ogT5FkLnSTMfX9kOaUh?p=preview

**Comment by wesleycho** [comment]
This might be a duplicate of #5190

**Comment by ricksmt** [comment]
I'm not quite sure what the digest part of the issue is, but I've also observed the `customClass` attribute not working as expected. I've finally overcome the issue with the attached decorator. I'll summarize my findings here:

- DatepickerPopup has two ways to supply the `customClass` function: either through `datepickerOptions` or a `custom-class` attribute.
- Supplying it via `datepickerOptions` copies the supplied function to the `custom-class` attribute on the generated Datepicker. AngularJS then throws a fit because it's expecting an expression, and AngularJS doesn't accept function declarations in expressions (specified [here](https://docs.angularjs.org/guide/expression)). (Line 2555 in ui-bootstrap-tpls.js.)
- Supplying it via the `custom-class` attribute results in no exceptions, but the supplied function is never called. (Line 2619, overrides line 2555's `custom-class`. Line 2071 then supposedly consumes the supplied function, but the function it consumes seems to return the supplied function.)

On a separate but related note: I'd appreciate it if the documentation was consistent (and correct). `customClass`is defined as an expression (that takes two parameters?), but then the example supplies a function that takes one object that holds the two parameters. I just noticed `dateDisabled` has a similar definition, so I suppose that would merit a review as well.

[DatepickerPopupDecorator.txt](https://github.com/angular-ui/bootstrap/files/162244/DatepickerPopupDecorator.txt)
